p2p/p2p_test.go

```diff
@@ -181,26 +196,76 @@ func TestSendRequest(t *testing.T) {
 		NoMdns:      true, // TODO: investigate failed dials, disable for now
 	}
 
-	msgSendB := make(chan Message)
-
-	nodeB := startNewService(t, configB, msgSendB, nil)
+	nodeB := startNewService(t, configB, nil, nil)
 	defer nodeB.Stop()
 
-	addrA := nodeA.host.fullAddrs()[0]
+	addrB := nodeB.host.fullAddrs()[0]
 
-	addrInfoA, err := peer.AddrInfoFromP2pAddr(addrA)
+	// Get address info for node A (used for `host.connect`)
+	addrInfoB, err := libp2pPeer.AddrInfoFromP2pAddr(addrB)
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	err = nodeB.host.connect(*addrInfoA)
+	err = nodeA.host.connect(*addrInfoB)
 	if err != nil {
 		t.Fatal(err)
 	}
 
+	// Wait for status exchange
+	time.Sleep(5 * time.Second)
```

Comment:
Lets chat some more about how to remove this

Reply:
Yes, please. Definitely not ideal for CI testing.